src/systems/physics/Physics.cc

```diff
@@ -3628,58 +3775,34 @@ void PhysicsPrivate::UpdateSim(EntityComponentManager &_ecm,
         return true;
       });
 
-  {
-    std::vector<Entity> entitiesJointVelocityCmd;
-    _ecm.Each<components::JointVelocityCmd>(
-        [&](const Entity &_entity, components::JointVelocityCmd *) -> bool
-        {
-          entitiesJointVelocityCmd.push_back(_entity);
-          return true;
-        });
-
-    for (const auto entity : entitiesJointVelocityCmd)
-    {
-      _ecm.RemoveComponent<components::JointVelocityCmd>(entity);
-    }
-  }
+  _ecm.Each<components::JointVelocityCmd>(
+      [&](const Entity &, components::JointVelocityCmd *_vel) -> bool
+      {
+        std::fill(_vel->Data().begin(), _vel->Data().end(), 0.0);
+        return true;
+      });
 
   _ecm.Each<components::SlipComplianceCmd>(
       [&](const Entity &, components::SlipComplianceCmd *_slip) -> bool
       {
         std::fill(_slip->Data().begin(), _slip->Data().end(), 0.0);
         return true;
       });
+  GZ_PROFILE_END();
 
-  {
-    std::vector<Entity> entitiesAngularVelocityCmd;
-    _ecm.Each<components::AngularVelocityCmd>(
-        [&](const Entity &_entity, components::AngularVelocityCmd *) -> bool
-        {
-          entitiesAngularVelocityCmd.push_back(_entity);
-          return true;
-        });
-
-    for (const auto entity : entitiesAngularVelocityCmd)
-    {
-      _ecm.RemoveComponent<components::AngularVelocityCmd>(entity);
-    }
-  }
-
-  {
-    std::vector<Entity> entitiesLinearVelocityCmd;
-    _ecm.Each<components::LinearVelocityCmd>(
-        [&](const Entity &_entity, components::LinearVelocityCmd *) -> bool
-        {
-          entitiesLinearVelocityCmd.push_back(_entity);
-          return true;
-        });
+  _ecm.Each<components::AngularVelocityCmd>(
+      [&](const Entity &, components::AngularVelocityCmd *_vel) -> bool
+      {
+        _vel->Data() = math::Vector3d::Zero;
+        return true;
+      });
 
-    for (const auto entity : entitiesLinearVelocityCmd)
-    {
-      _ecm.RemoveComponent<components::LinearVelocityCmd>(entity);
-    }
-  }
-  GZ_PROFILE_END();
```

Comment:
this appears to reverse some changes from https://github.com/gazebosim/gz-sim/pull/2228; can you revert that so the `*VelocityCmd` handling is unchanged?

Reply:
yeah, I didn't notice this sorry for that. I have reverted and made the required changes.